testbeam_analysis/tools/kalman.py

```diff
@@ -152,12 +158,63 @@ def _filter(transition_matrices, observation_matrices, transition_covariances,
     kalman_gains = np.zeros((chunk_size, n_timesteps, n_dim_state, n_dim_obs))
     filtered_states = np.zeros((chunk_size, n_timesteps, n_dim_state))
     filtered_state_covariances = np.zeros((chunk_size, n_timesteps, n_dim_state, n_dim_state))
+    # array where new transition matrices are stored, needed to pass it to kalman smoother
+    transition_matrices_update = np.zeros_like(transition_covariances)
 
     for t in range(n_timesteps):
         if t == 0:
             predicted_states[:, t] = initial_state
```

Comment:
The rotation of first plane also matters. But in the first step we cant do a prediction, thus the filtered state is the inital state. After this 'initialization' we predict the state to the second plane. To include rotation we calculate the intersection of the filtered state with the first plane (therefore rotation of first plane is considered) and the next plane.